### ISSUE TITLE
Closing a modal window using Mono/Gtk crashes

### ISSUE THREAD

**Issue by Mafii** [AvaloniaUI/Avalonia#729]
IDPA-2016-NEAT-CNN/NEAT_Visualizer#5 explains the steps to reproduce.

It doesn't happen on Windows using .NET (Migrosoft) but using Gtk on Windows, it does.

Error seems to happen somewhere in Gtk.

**Comment by Mafii** [comment]
Remark - works with Mono and Direct2D

**Comment by grokys** [comment]
Seems to happen because `Avalonia.Gtk.WindowImplBase.Invalidate` is called after window has been disposed.

**Comment by Silverfeelin** [comment]
I wasn't sure if I should open a new issue, but I'm running into the same problem in using Avalonia Beta 1. The gist has been removed from the linked issue, but based on the description it seems like we're doing the same thing.

I have created a repository with the minimal code needed to reproduce the issue.  
https://github.com/Silverfeelin/AvaloniaCloseShowDialog

What I'm trying to do is simply use a button click event to close the dialog window (in my actual project I'm using this to notify the user of input errors).

On Windows 10, closing the dialog works perfectly fine. On Linux (Ubuntu 16.04), the following exception occurs when Close() is called from within the dialog code:

```
Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Avalonia.Gtk3.WindowBaseImpl.StaticTickCallback(IntPtr widget, IntPtr clock, IntPtr userData)
   at Avalonia.Gtk3.Gtk3Platform.RunLoop(CancellationToken cancellationToken)
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken)
   at AvaloniaCloseShowDialog.Program.Main(String[] args) in C:\Users\Silver\source\repos\AvaloniaCloseShowDialog\AvaloniaCloseShowDialog\Program.cs:line 11
```

---

Related code in the repository linked above:

Opening dialog window: https://github.com/Silverfeelin/AvaloniaCloseShowDialog/blob/master/AvaloniaCloseShowDialog/MainWindow.xaml.cs#L21-L25

Closing dialog window:
https://github.com/Silverfeelin/AvaloniaCloseShowDialog/blob/3fb0cb81f7b388e17158b3003840654feea0508e/AvaloniaCloseShowDialog/Dialog.xaml.cs#L22-L25

Apologies in advance if this is a mistake on my end.
